src/main/scala/beam/agentsim/agents/vehicles/BeamVehicle.scala

```diff
@@ -143,6 +146,23 @@ class BeamVehicle(
     }
   }
 
+  def waitingToCharge(startTick: Int): Unit = {
```

Comment:
Can you rename the method name or the variable name? Having the same name as a variable and method is confusing.